docs/source/tutorial.ipynb

```diff
@@ -3687,6 +3687,294 @@
+       "<tr><td><i style='opacity: 0.6'>2</i></td><td style=\"text-align: right;\">  2</td><td>True      </td></tr>\n",
+       "<tr><td><i style='opacity: 0.6'>3</i></td><td style=\"text-align: right;\">  3</td><td>True      </td></tr>\n",
+       "<tr><td><i style='opacity: 0.6'>4</i></td><td style=\"text-align: right;\">  4</td><td>False     </td></tr>\n",
+       "<tr><td><i style='opacity: 0.6'>5</i></td><td style=\"text-align: right;\">  5</td><td>True      </td></tr>\n",
+       "<tr><td><i style='opacity: 0.6'>6</i></td><td style=\"text-align: right;\">  6</td><td>False     </td></tr>\n",
+       "<tr><td><i style='opacity: 0.6'>7</i></td><td style=\"text-align: right;\">  7</td><td>True      </td></tr>\n",
+       "<tr><td><i style='opacity: 0.6'>8</i></td><td style=\"text-align: right;\">  8</td><td>False     </td></tr>\n",
+       "<tr><td><i style='opacity: 0.6'>9</i></td><td style=\"text-align: right;\">  9</td><td>False     </td></tr>\n",
+       "</tbody>\n",
+       "</table>"
+      ],
+      "text/plain": [
+       "  #    x  is_prime\n",
+       "  0    0  False\n",
+       "  1    1  False\n",
+       "  2    2  True\n",
+       "  3    3  True\n",
+       "  4    4  False\n",
+       "  5    5  True\n",
+       "  6    6  False\n",
+       "  7    7  True\n",
+       "  8    8  False\n",
+       "  9    9  False"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import vaex\n",
+    "\n",
+    "def slow_is_prime(x):\n",
+    "    return x > 1 and all((x % i) != 0 for i in range(2, x))\n",
+    "\n",
+    "df = vaex.from_arrays(x=vaex.vrange(0, 100_000, dtype='i4'))\n",
+    "# you need to explicitly tell which arguments you need\n",
+    "df['is_prime'] = df.apply(slow_is_prime, arguments=[df.x])\n",
+    "df.head(10)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2020-11-27T18:59:25.178658Z",
+     "start_time": "2020-11-27T18:59:23.881404Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "There are 9592 prime number between 0 and 100000\n"
+     ]
+    }
+   ],
+   "source": [
+    "prime_count = df.is_prime.sum()\n",
+    "print(f'There are {prime_count} prime number between 0 and {len(df)}')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2020-11-27T18:59:25.181779Z",
+     "start_time": "2020-11-27T18:59:25.179695Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# both of these are equivalent\n",
+    "df['is_prime'] = df.apply(slow_is_prime, arguments=[df.x])\n",
+    "# but this form only works for a single argument\n",
+    "df['is_prime'] = df.x.apply(slow_is_prime)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## When not to use apply\n",
```

Comment:
```suggestion
    "## When not to use `apply`\n",
```

Reply:
this never format correctly with nbsphinx